### DISCUSSION TITLE
Exploring GoCD for bitbucket pipeline replacement

### DISCUSSION THREAD

**Discussion by vigneshraaj** [gocd/gocd#9638] [Q&A] [answered]
Hi. I’m not sure if this is the right forum for it, if not please direct me to right forum.
We are exploring GoCD for our CI/CD pipeline. Currently we are using Bitbucket Pipelines.
I tried locally and have few questions regarding how-to,
1. Can we configure a pipeline without git dependency / material? We have a pipeline which runs on a schedule and the pipeline is just pure shell script (and it does not depend on any code from any repo).
2. How to configure multi-branch pipeline? For example, we have an frontend repo with different branch for different environment, like dev, prod, … so that when triggering the pipeline manually, we can select which branch to trigger from.
3. Here is a complex scenarios which we trying to resolve. We have 2 repo, 1 for frontend and 2 for backend. The backend pipeline will be dependent on frontend pipeline. Whenever the frontend pipeline build is successful, it should trigger backend pipeline. Also if there is any change in backend repo, it should trigger backend pipeline. I’m able to configure this by adding multiple dependencies for backend pipeline. The complexity here is, we generated random key when building frontend (For eg.: KEY=hash-BUILD-1), and the same needs to be passed to backend pipeline. 
	a. How to trigger backend pipeline with custom build parameter (KEY)? From pipeline 1 it should pass parameter and value for pipeline 2 build. 
	b. How to persist param value (KEY) in backend pipeline, so that subsequent build from backend repo (commits, poll scm) trigger will use the param value (KEY) which is already set. Param value changes only when frontend pipeline creates a new build.

**Comment by chadlwilson** [comment, marked as answer]
I believe you got some feedback/ideas at https://gitter.im/gocd/gocd?at=6124fb5c63dca81891209b74 and https://gitter.im/gocd/gocd?at=61264487c1d4ec13466f301c :)
